src/pkg/clients/Service_discovery.go

```diff
@@ -111,6 +114,14 @@ func (s *serviceDiscovery) registerService(serviceInfos ServiceInfos) error {
 	return nil
 }
 
+func (s *serviceDiscovery) GetExistingServices() []models.TransformerService {
+	existingServices := []models.TransformerService{}
+	for name := range s.transformersAddressesList {
+		existingServices = append(existingServices, *models.CreateTransformerService(name))
+	}
+	return existingServices
```

Comment:
Since transformersAddressesList can be modified by another goroutine, you need a mutex to access it (RLock and RUnlock)